File: ios/Sources/CapacitorWorkoutsConnectedAppsPlugin/CapacitorWorkoutsConnectedAppsPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorWorkoutsConnectedAppsPlugin)
public class CapacitorWorkoutsConnectedAppsPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CapacitorWorkoutsConnectedAppsPlugin"
    public let jsName = "CapacitorWorkoutsConnectedApps"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = CapacitorWorkoutsConnectedApps()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    @objc func isConnectedAppsEnabled(_ call: CAPPluginCall) {
        let isEnabled = implementation.isConnectedAppsEnabled()
        // more logic here
        call.resolve([
            "isConnectedAppsEnabled": isEnabled
        ])
    }

    @objc func saveWorkoutToConnectedApps(_ call: CAPPluginCall) {
        let workout = call.getString("workout") ?? ""
        let isSaved = implementation.saveWorkoutToConnectedApps(workout: workout)
        // more logic here
        call.resolve([
            "savedSuccessfully": isSaved
        ])
    }
}

File: android/src/main/java/com/egym/plugins/capacitorworkoutsconnectedappsplugin/CapacitorWorkoutsConnectedAppsPlugin.java
package com.egym.plugins.capacitorworkoutsconnectedappsplugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "CapacitorWorkoutsConnectedApps")
public class CapacitorWorkoutsConnectedAppsPlugin extends Plugin {

    private CapacitorWorkoutsConnectedApps implementation = new CapacitorWorkoutsConnectedApps();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public isConnectedAppsEnabled(PluginCall call) {
        boolean isEnabled = implementation.isConnectedAppsEnabled();
        JSObject ret = new JSObject();
        ret.put("isEnabled", isEnabled);

        //implement more logic here

        call.resolve(ret);
    }

    @PluginMethod
    public saveWorkoutToConnectedApps(PluginCall call) {
        JSObject workout = call.getObject("workout");
        if (!call.getData().has("id")) {
            call.reject("Must provide an id");
            return;
        }

        //implement more logic here

        JSObject ret = new JSObject();
        ret.put("savedSuccessfully", true);
        call.resolve(ret);
    }
    
}
